aci-preupgrade-validation-script.py

```diff
@@ -2677,6 +2677,42 @@ def sup_hwrev_check(index, total_checks, cversion, tversion, **kwargs):
     return result
 
 
+def uplink_limit_check(index, total_checks, cversion=None, tversion=None, **kwargs):
+    title = 'Pre-Leaf Fabric Uplink Limit Validation'
+    result = PASS
+    msg = ''
+    headers = ["Node", "Uplink Count"]
+    data = []
+    recommended_action = "Reduce Per-Leaf Port Profile Uplinks to supported scale; 56 or less."
+    doc_url = 'http://cs.co/ACI_Access_Interfaces_Config_Guide'
+
+    print_title(title, index, total_checks)
+
+    cfw = AciVersion(cversion)
+    tfw = AciVersion(tversion)
+
+    if cfw.older_than("6.0(1a)") and tfw.newer_than("6.0(1a)"):
+        port_profiles = icurl('class', 'eqptPortP.json?query-target-filter=eq(eqptPortP.ctrl,"uplink")')
+        if not port_profiles or (len(port_profiles) < 57):
+            return result
+
+        node_count = {}
+        for pp in port_profiles:
+            dn = re.search(node_regex, pp['eqptPortP']['attributes']['dn'])
+            node_id = dn.group("node")
+            node_count.setdefault(node_id, 0)
+            node_count[node_id] += 1
+
+        for node, count in node_count.items():
```

Comment:
You are using the last `node_id` from the previous iteration of `for pp in port_profiles:` instead of the node ID that meets the condition.

```suggestion
        for node_id, count in node_count.items():
```

Reply:
done